api/client/client.go

```diff
@@ -111,19 +108,18 @@ type Config struct {
 	// KeepAliveCount specifies the amount of missed keep alives
 	// to wait for before declaring the connection as broken
 	KeepAliveCount int
-	// TLS is the client's TLS config
-	TLS *tls.Config
+	// Credentials are used to authenticate the client's connection to the server
+	Credentials
 }
 
 // CheckAndSetDefaults checks and sets default config values
 func (c *Config) CheckAndSetDefaults() error {
+	if err := c.Credentials.CheckAndSetDefaults(); err != nil {
+		return trace.WrapWithMessage(err, "invalid client credentials")
```

Comment:
nit: `WrapWithMessage` is not necessary - it's a longer wording of the `Wrap` API. No idea why it was introduced